hypothesis-python/docs/compatibility.rst

```diff
@@ -25,80 +22,71 @@ changes in patch releases.
-test to take this form, or ask the framework maintainer to support our
-:ref:`hooks for inserting such a wrapper later <custom-function-execution>`.
-
-In terms of what's actually *known* to work:
-
-* Hypothesis integrates as smoothly with pytest and unittest as we can make it,
-  and this is verified as part of the CI.
-* :pypi:`pytest` fixtures work in the usual way for tests that have been decorated
-  with |@given| - just avoid passing a strategy for
-  each argument that will be supplied by a fixture.  However, function-scoped fixtures
-  will run only once for the whole function, not per example. To proactively warn you about
-  this case, we raise |HealthCheck.function_scoped_fixture|, unless suppressed with
-  |settings.suppress_health_check|.
-* The :func:`python:unittest.mock.patch` decorator works with
-  |@given|, but we recommend using it as a context
-  manager within the decorated test to ensure that the mock is per-test-case
-  and avoid poor interactions with Pytest fixtures.
-* Nose works fine with Hypothesis, and this is tested as part of the CI. ``yield`` based
-  tests simply won't work.
-* Integration with Django's testing requires use of the :ref:`hypothesis-django` extra.
-  The issue is that in Django's tests' normal mode of execution it will reset the
-  database once per test rather than once per example, which is not what you want.
-* :pypi:`coverage` works out of the box with Hypothesis; our own test suite has
-  100% branch coverage.
+In general, Hypothesis goes to quite a lot of effort to return a function from |@given| that behaves as closely to a normal test function as possible. This means that most things should work sensibly with most testing frameworks.
+
+Maintainers of testing frameworks may be interested in our support for :ref:`custom function execution <custom-function-execution>`, which may make some Hypothesis interactions possible to support.
+
+pytest
+~~~~~~
+
+The main interaction to be aware of between Hypothesis and :pypi:`pytest` is fixtures.
+
+pytest fixtures are automatically passed to |@given| tests, as usual. Note that |@given| supplies parameters from the right, so tests which use a fixture should be written with the fixture placed first:
+
+.. code-block:: python
+
+  @given(st.integers())
+  def test_use_fixture(myfixture, n):
+      pass
+
+However, function-scoped fixtures run only once for the entire test, not per-input. This can be surprising for fixtures which are expected to set up per-input state. To proactively warn about this, we raise |HealthCheck.function_scoped_fixture| (unless suppressed with |settings.suppress_health_check|).
+
+unittest
+~~~~~~~~
+
+:pypi:`unittest` works out of the box with Hypothesis.
+
+The :func:`python:unittest.mock.patch` decorator works with |@given|, but we recommend using it as a context manager within the test instead, to ensure that the mock is per-input, and to avoid poor interactions with Pytest fixtures.
```

Comment:
https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests is a noop with Hypothesis (we patch it, iirc)